test/dlblob_test.js

```diff
@@ -0,0 +1,201 @@
+                    resolve(mock.getAuthResp());
+                });
+        });
+        
+        // Expected Alert Logic parameters
+        process.env.WEBSITE_HOSTNAME = 'app-name';
+        process.env.WEBSITE_SITE_NAME = 'test-site';
+        process.env.CUSTOMCONNSTR_APP_AL_ACCESS_KEY_ID = mock.AL_KEY_ID;
+        process.env.CUSTOMCONNSTR_APP_AL_SECRET_KEY = mock.AL_SECRET;
+        process.env.CUSTOMCONNSTR_APP_AL_API_ENDPOINT = mock.AL_API_ENDPOINT;
+        process.env.CUSTOMCONNSTR_APP_AL_RESIDENCY = 'default';
+        process.env.APP_INGEST_ENDPOINT = 'existing-ingest-endpoint';
+        process.env.APP_AZCOLLECT_ENDPOINT = 'existing-azcollect-endpoint';
+        process.env.COLLECTOR_HOST_ID = 'existing-host-id';
+        process.env.COLLECTOR_SOURCE_ID = 'existing-source-id';
+        process.env.APP_DL_CONTAINER_NAME = 'alertlogic-dl';
+        process.env.DL_BLOB_PAGE_SIZE = '100';
+        
+        // Expected Azure parameters
+        process.env.APP_SUBSCRIPTION_ID = 'subscription-id';
+        process.env.APP_RESOURCE_GROUP = 'kktest11-rg';
+        process.env.APP_TENANT_ID = 'tenant-id';
+        process.env.CUSTOMCONNSTR_APP_CLIENT_ID = 'client-id';
+        process.env.CUSTOMCONNSTR_APP_CLIENT_SECRET = 'client-secret';
+        process.env.AzureWebJobsStorage = 'DefaultEndpointsProtocol=https;AccountName=kktestdl;AccountKey=S0meKey+';
+        
+        processLogStub = sinon.stub(AlAzureCollector.prototype, 'processLog').callsFake(
+                function fakeFn(message, formatFun, hostmetaElems, callback) {
+                    formatFun(message);
+                    return callback(null);
+                });
+    });
+    
+    after(function() {
+        nock.restore();
+        fakeAuth.restore();
+        processLogStub.restore();
+    });
+
+    beforeEach(function() {
+        if (!nock.isActive()) {
+            nock.activate();
+        }
+        processLogStub.resetHistory();
+    });
+    
+    afterEach(function() {
+        nock.cleanAll();
+    });
+    
+    it('Simple OK check', function(done) {
+        process.env.DL_BLOB_PAGE_SIZE = '10';
+        // Mock Azure HTTP calls
+        // List blobs
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
+        .get('/alertlogic-dl')
+        .query({'restype':'container','comp':'list','maxresults':process.env.DL_BLOB_PAGE_SIZE,'prefix':process.env.WEBSITE_SITE_NAME})
+        .times(5)
+        .reply(200, mock.LIST_CONTAINER_BLOBS());
+        
+        // Get blob content
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
+        .get(/alertlogic-dl.*/)
+        .times(6)
+        .reply(200, mock.GET_BLOB_CONTENT_TEXT);
+        
+        // Delete blob
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
+        .delete(/alertlogic-dl.*/)
+        .times(6)
+        .reply(202);
```

Comment:
This doesn't check that delete was actually called, just returns 202 6 times if called, right? If so, maybe worth adding a check that delete was actually called (6 times)?